### PR TITLE
vscode-extensions.matklad.rust-analyzer: fix on Darwin

### PR DESCRIPTION
**Pull request by kubukoz**
###### Motivation for this change

The package doesn't build on aarch64-darwin (Monterey).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by antifuchs — APPROVED** [review]
Yep, this seems like the standard way to do it.

**Comment by winterqt** [comment]
Sadly this is still an issue with this patch on aarch64-darwin. This definitely is related to the fact that I have the sandbox enabled, but I'm not sure if we can even fix this in Nixpkgs...

```
> keytar@7.7.0 install /nix/store/vqqkyj3zvl4pq5z1njrb2kmhmm69g1mi-rust-analyzer-0.2.834/lib/node_modules/rust-analyzer/node_modules/keytar
> prebuild-install || npm run build

sh: line 1: prebuild-install: command not found

> keytar@7.7.0 build /nix/store/vqqkyj3zvl4pq5z1njrb2kmhmm69g1mi-rust-analyzer-0.2.834/lib/node_modules/rust-analyzer/node_modules/keytar
> node-gyp rebuild

Traceback (most recent call last):
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1288, in XcodeVersion
    raise GypError("xcodebuild returned unexpected results")
gyp.common.GypError: xcodebuild returned unexpected results

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py", line 50, in <module>
    sys.exit(gyp.script_main())
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 554, in script_main
    return main(sys.argv[1:])
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 547, in main
    return gyp_main(args)
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 532, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 2214, in GenerateOutput
    writer.Write(qualified_target, base_path, output_file, spec, configs,
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 818, in Write
    self.WriteSources(
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/generator/make.py", line 1218, in WriteSources
    cflags = self.xcode_settings.GetCflags(configname)
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 565, in GetCflags
    archs = self.GetActiveArchs(self.configname)
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 427, in GetActiveArchs
    xcode_archs_default = GetXcodeArchsDefault()
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 122, in GetXcodeArchsDefault
    xcode_version, _ = XcodeVersion()
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1292, in XcodeVersion
    version = CLTVersion()  # macOS Catalina returns 11.0.0.0.1.1567737322
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1320, in CLTVersion
    output = GetStdout(['/usr/sbin/pkgutil', '--pkg-info', key])
  File "/nix/store/7q04aq0sq6im9a0k09gzfa1xfncc0xgm-nodejs-14.18.1/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1342, in GetStdout
    job = subprocess.Popen(cmdlist, stdout=subprocess.PIPE)
  File "/nix/store/26ki0p1nk7hibkdn71z90ihzr5kxdm71-python3-3.9.6/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/26ki0p1nk7hibkdn71z90ihzr5kxdm71-python3-3.9.6/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
PermissionError: [Errno 1] Operation not permitted: '/usr/sbin/pkgutil'
```

**Comment by kubukoz** [comment]
huh, checked again and indeed, it's failing in sandbox :(

File: pkgs/misc/vscode-extensions/rust-analyzer/default.nix [hunk 1]
```diff
@@ -37,9 +39,15 @@ let
...
+      jq
+      moreutils
+      esbuild
       # Required by `keytar`, which is a dependency of `vsce`.
-      pkg-config libsecret
+      pkg-config
+      libsecret
+    ] ++ lib.optional stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+      darwin.apple_sdk.frameworks.AppKit
```
Comment (SuperSandro2000) [inline pkgs/misc/vscode-extensions/rust-analyzer/default.nix:50]:
```suggestion      
      AppKit
      Security
```
and add those to inputs and inherit the frameworks in the top-level call.

**Comment by SuperSandro2000** [comment]
> huh, checked again and indeed, it's failing in sandbox :(

Darwin sandbox is always a bit tricky and sometimes half broken. The frameworks are missing anyway and this can be merged without it working in sandbox.

File: pkgs/misc/vscode-extensions/rust-analyzer/default.nix (hunk 1 above)
Comment (kubukoz) [reply pkgs/misc/vscode-extensions/rust-analyzer/default.nix]: Not 100% sure I did what you meant, please take a look :)

**Comment by kubukoz** [comment]
Update, this has been done in https://github.com/NixOS/nixpkgs/pull/152286.

**Pull request closed without merging**